internal/clients/provider_config.go

```diff
@@ -264,9 +264,14 @@ func UseProviderSecret(ctx context.Context, data []byte, profile, region string)
 // AssumeRoleWithWebIdentity & AssumeRoles.
 func GetRoleChainConfig(ctx context.Context, pcs *v1beta1.ProviderConfigSpec, cfg *aws.Config) (*aws.Config, error) {
 	pCfg := cfg
+	regionOpt := func(o *sts.Options) {
+		if cfg.Region == "" {
+			o.Region = GlobalRegion
+		}
+	}
```

Comment:
might be nice to pull this out into something like:
```go
NewRegionFallbackOpt(region string) func(*sts.Options) {
  if region != "" {
    return func(*sts.Options) {}
  }
  return func(o *sts.Options) {
    o.Region = GlobalRegion
  }
}
```

But I am fine with deferring doing so for now.